### PR TITLE
test: Proto Column Integration tests

### PR DESCRIPTION
**Pull request by gauravpurohit06**
This PR includes Integration tests for following scenarios: 
-  Write & Read Proto Messages & Enums and array of it. 
- Check compatibility between ProtoMessage/Bytes & ProtoEnun/Int64. 
- Negative Case when incorrect Proto object is provided while reading the data.
-   Parameterised queries, Primary Keys and Indexes.

### CODE REVIEW

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITProtoColumnTest.java [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+public class ITProtoColumnTest {
+  private static String projectId;
+  private static String instanceId;
+  private static String databaseId;
+  private static DatabaseClient databaseClient;
+
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    assumeFalse(
+        "Proto Column is not supported in the emulator", EmulatorSpannerHelper.isUsingEmulator());
```
Comment (harshachinta) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITProtoColumnTest.java:65]: Do we need this here as it is also mentioned inside `@Test` methods below?

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITProtoColumnTest.java [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+        Arrays.asList(
+            singerInfoBytes,
+            null,
+            ByteArray.copyFrom(SingerInfo.getDefaultInstance().toByteArray()));
+
+    List<ProtocolMessageEnum> enumList = Arrays.asList(Genre.FOLK, null, Genre.ROCK);
+    List<Long> enumConstList =
+        Arrays.asList((long) Genre.FOLK_VALUE, null, (long) Genre.ROCK_VALUE);
+
+    databaseClient.write(
```
Comment (harshachinta) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITProtoColumnTest.java:122]: Check compatibility between Proto Messages & Bytes, Proto Enum & Int64 during write operation also

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITProtoColumnTest.java [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+                          "INSERT INTO Singers (SingerId, FirstName, LastName, SingerInfo, SingerGenre) VALUES (1, \"FirstName1\", \"LastName1\", @singerInfo, @singerGenre)")
+                      .bind("singerInfo")
+                      .to(singerInfo1)
+                      .bind("singerGenre")
+                      .to(genre1)
+                      .build();
+
+              Statement statement2 =
+                  Statement.newBuilder(
+                          "INSERT INTO Singers (SingerId, FirstName, LastName, SingerInfo, SingerGenre) VALUES (2, \"FirstName2\", \"LastName2\", @singerInfo, @singerGenre)")
```
Comment (harshachinta) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITProtoColumnTest.java:222]: Same as above SingerId value

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITProtoColumnTest.java (hunk 1 above)
Comment (gauravpurohit06) [reply google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITProtoColumnTest.java:65]: It's good to have it at both the places as `beforeClass` will be executed prior to test.  Going forward if we have some test which are supported we can remove it from here.

**Review by harshachinta — APPROVED** [review]
LGTM

**Pull request merged**